pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -169,6 +171,32 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
 		return nil, err
 	}
 
+	// Replace the standard k8s emptyDir volume plugin with a wrapper version
+	// which offers XFS quota functionality, but only if the node config
+	// specifies an empty dir quota to apply to projects:
+	if options.ProjectEmptyDirQuota != nil {
+		glog.V(2).Info("Replacing empty-dir volume plugin with quota wrapper")
+		wrappedEmptyDirPlugin := false
+		for idx, plugin := range cfg.VolumePlugins {
+			// Can't really do type checking or use a constant here as they are not exported:
+			if plugin.Name() == "kubernetes.io/empty-dir" {
```

Comment:
How do folks feel about this as a compromise solution? ^^ This saves us having to go back and reworking and re-testing (as we're hoping to get this in for online launch), avoids the rebase breakage problem, and as ever we can hopefully take a crack and adapting it directly into the upstream API later.